test/integration/use_int_test.go

```diff
@@ -158,37 +143,67 @@ func (suite *UseIntegrationTestSuite) TestReset() {
 	if runtime.GOOS != "windows" {
 		suite.NoError(err)
 		suite.Contains(string(fileutils.ReadFileUnsafe(rcfile)), ts.Dirs.DefaultBin, "PATH does not have default project in it")
-		suite.Contains(string(fileutils.ReadFileUnsafe(zshRcFile)), ts.Dirs.DefaultBin, "PATH does not have default project in it")
 	}
 
-	cp = ts.SpawnWithOpts(
-		e2e.WithArgs("use", "reset"),
-		e2e.AppendEnv("SHELL=bash"),
-	)
+	cp = ts.SpawnWithOpts(e2e.WithArgs("use", "reset"))
 	cp.Expect("Continue?")
 	cp.SendLine("n")
 	cp.Expect("Reset aborted by user")
 	cp.ExpectExitCode(1)
 
-	cp = ts.SpawnWithOpts(
-		e2e.WithArgs("use", "reset", "--non-interactive"),
-		e2e.AppendEnv("SHELL=bash"),
-	)
+	cp = ts.SpawnWithOpts(e2e.WithArgs("use", "reset", "--non-interactive"))
 	cp.Expect("Reset default project runtime")
 	cp.Expect("Note you may need to")
 	cp.ExpectExitCode(0)
 
 	suite.False(fileutils.TargetExists(python3Exe), python3Exe+" still exists")
 
+	cp = ts.SpawnWithOpts(e2e.WithArgs("use", "reset", "-n"))
+	cp.Expect("No global default project to reset")
+	cp.ExpectExitCode(0)
+
+	if runtime.GOOS != "windows" {
+		suite.NotContains(string(fileutils.ReadFileUnsafe(rcfile)), ts.Dirs.DefaultBin, "PATH still has default project in it")
+	}
+}
+
+func (suite *UseIntegrationTestSuite) TestUseShellUpdates() {
+	suite.OnlyRunForTags(tagsuite.Use)
```

Comment:
Let's move this to shell_int_test.go and use `tagsuite.Shell`. "state use" isn't what we're testing here, it's just what we're using to facilitate the test.